src/modules/adaptors/Morpho/MorphoBlue/MorphoBlueHelperLogic.sol

```diff
@@ -0,0 +1,111 @@
+
+    constructor(address _morphoBlue) {
+        morphoBlue = IMorpho(_morphoBlue);
+    }
+
+    /**
+     * @notice The ```_getHealthFactor``` function returns the current health factor of a respective position given an exchange rate.
+     * @param _id The specified Morpho Blue market Id.
+     * @param _market The specified Morpho Blue market.
+     * @return currentHF The health factor of the position atm.
+     */
+    function _getHealthFactor(Id _id, MarketParams memory _market) internal view virtual returns (uint256 currentHF) {
+        uint256 borrowAmount = _userBorrowBalance(_id, address(this));
+        if (borrowAmount == 0) return type(uint256).max;
+        uint256 collateralPrice = IOracle(_market.oracle).price(); // recall from IOracle.sol that the units will be 10 ** (36 - collateralUnits + borrowUnits) BUT collateralPrice is in units of borrow.
+
+        // get collateralAmount in borrowAmount for LTV calculations
+        uint256 collateralAmount = _userCollateralBalance(_id, address(this));
+        console.log(
+            "userCollateralBalance: %s, collateralPrice: %s, collateralAmount: %s",
+            _userCollateralBalance(_id, address(this)),
+            collateralPrice,
+            collateralAmount
+        );
+        uint256 collateralAmountInBorrowUnits = collateralAmount.mulDivDown(collateralPrice, ORACLE_PRICE_SCALE);
+
+        if (collateralAmountInBorrowUnits == 0) return 0;
+        uint256 LTV_PRECISION = 1e18;
+        uint256 currentPositionLTV = borrowAmount.mulDivUp(LTV_PRECISION, collateralAmountInBorrowUnits);
+        // convert LTVs to HF
+        currentHF = (_market.lltv).mulDivDown(1e18, currentPositionLTV);
+    }
+
+    /**
+     * @dev helper function that returns actual supply position amount for specified `_user` according to MB market accounting. This is alternative to using the MB periphery libraries that simulate accrued interest balances.
+     * @param _id identifier of a Morpho Blue market.
+     * @param _user address that this function will query Morpho Blue market for.
+     * @return Actual supply amount for the `_user`
+     * NOTE: make sure to call `accrueInterest()` on respective market before calling these helpers.
+     */
+    function _userSupplyBalance(Id _id, address _user) internal view returns (uint256) {
+        Market memory market = morphoBlue.market(_id);
+        return (morphoBlue.supplyShares(_id, _user).toAssetsDown(market.totalSupplyAssets, market.totalSupplyShares));
+    }
+
+    /**
+     * @dev helper function that returns actual collateral position amount for specified `_user` according to MB market accounting. This is alternative to using the MB periphery libraries that simulate accrued interest balances.
+     * @param _id identifier of a Morpho Blue market.
+     * @param _user address that this function will query Morpho Blue market for.
+     * NOTE: make sure to call `accrueInterest()` on respective market before calling these helpers.
```

Comment:
This is not required for collateral as collateral does not accrue interest

Reply:
Thanks! Removed the nat spec that does not apply.